sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs

```diff
@@ -72,13 +72,16 @@ internal class ReceiverManager
             CancellationToken cancellationToken,
             bool forceClose = false)
         {
-            try
+            if (Receiver != null)
             {
-                await Receiver.DisposeAsync().ConfigureAwait(false);
-            }
-            finally
-            {
-                Receiver = null;
+                try
+                {
+                    await Receiver.DisposeAsync().ConfigureAwait(false);
```

Comment:
We probably should go with `await Receiver?.DisposeAsync().ConfigureAwait(false);` over the `if` check, as I believe we'd still have a race condition otherwise, no?

Reply:
Yeah, that could definitely happen to somebody. I'm trying to find a good way to write this though, everything I come up with looks really horrifying. Here is my current idea, which I'm not crazy about:

```csharp
await (Receiver?.DisposeAsync()).GetValueOrDefault().ConfigureAwait(false);
```

Reply:
could also do this I guess:
```csharp
var capturedReceiver = Receiver;
if (capturedReceiver != null)
{
    try
    {
        await capturedReceiver.DisposeAsync().ConfigureAwait(false);
    }
    finally
    {
        Receiver = null;
    }
}
```

Reply:
The latter is probably the more readable approach, IMHO.